src/main/java/com/example/codebase/domain/magazine/service/MagazineCategoryService.java

```diff
@@ -90,26 +90,35 @@ public MagazineCategoryResponse.Get updateCategory(Long categoryId, MagazineCate
 
         MagazineCategory parentCategory = findParentCategory(request.getParentId());
 
-        if (parentCategory != null) {
-            parentCategory.checkDepth();
-        }
+        validateParentCategory(category, parentCategory);
 
-        checkCategoryExists(request, parentCategory);
+        checkCategoryExists(request, category, parentCategory);
         category.changeParentCategory(parentCategory);
 
         category.update(request);
         magazineCategoryRepository.save(category);
         return MagazineCategoryResponse.Get.from(category);
     }
 
-    private void checkCategoryExists(MagazineCategoryRequest.Update request, MagazineCategory parentCategory) {
-        boolean exists = magazineCategoryRepository.existsByNameAndParent(request.getName(), parentCategory);
+    private void validateParentCategory(MagazineCategory category, MagazineCategory parentCategory) {
```

Comment:
```suggestion
    private void validateParentCategory(MagazineCategory category, MagazineCategory parentCategory) {
        if (parentCategory != null) {
            if (parentCategory.equals(category)) {
                throw new RuntimeException("부모 카테고리로 자기 자신을 참조할 수 없습니다.");
            }

            parentCategory.checkDepth();
        }
    }
```
반복문 체크 전에 자기자신 체크하면 좋을듯

Reply:
반복문 전에 자기 자신을 체크하도록 순서를 변경했습니다.  [0a325b7](https://github.com/Media-XI/artscope-backend/pull/165/commits/0a325b7cf9d7cf8f3f73536406718c150c8fc442)